src/dbnode/encoding/series_iterator_accumulator_test.go

```diff
@@ -242,29 +235,22 @@ func assertTestSeriesAccumulatorIterator(
 	}
 	// Ensure further calls to next false
 	for i := 0; i < 2; i++ {
-		assert.Equal(t, false, iter.Next())
+		assert.False(t, iter.Next())
 	}
 	if series.expectedErr == nil {
 		assert.NoError(t, iter.Err())
 	} else {
 		assert.Equal(t, series.expectedErr.err, iter.Err())
 	}
 
-	var tagIter ident.TagIterator
-	if series.retainTag {
-		checkTags()
-		tagIter = iter.Tags()
-	} else {
-		assert.Nil(t, iter.Tags())
-	}
-
 	assert.Equal(t, series.id, iter.id.String())
 	assert.Equal(t, series.nsID, iter.nsID.String())
 	iter.Close()
-	if series.retainTag {
-		// Check that the tag iterator was closed.
-		assert.False(t, tagIter.Next())
-	}
+
+	// Check that the tag iterator was closed.
+	tagIter := iter.Tags()
+	require.NotNil(t, tagIter)
+	assert.False(t, tagIter.Next())
```

Comment:
I think this might not be relevant anymore. IIUC, previously it was checking if the copied tag iterator was being closed.

https://github.com/m3db/m3/blob/fb084bc9484340effbf414eb108a3afe59bd0af8/src/dbnode/encoding/series_iterator_accumulator.go#L187-L190

But this PR removes the latter.

Reply:
My goal here was to make sure that after `Close` `Tags` returns `ident.EmptyTagIterator` and not `nil` (which I believe was also the intended behaviour in the original implementation). Getting `nil` from `Tags` is really ugly and may be causing panics in very distant pieces of code (such as the one provided in PR description), and the source of the error is really difficult to locate.